### PR TITLE
executor: fix point/batch on old row format clustered primary key with prefix columns

### LINKED ISSUE #23646: pointUpdate clustered index with old row format maybe miss remove old index contains prefix columns
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

need run with tikv

set `set global tidb_row_format_version=1` and reconnect

```
drop table if exists txx;
create table txx(c1 varchar(100), c2 set('dav', 'aaa'), c3 varchar(100), primary key(c1(2), c2) clustered, unique key uk1(c2), index idx1(c2, c1, c3));
insert into txx select 'AarTrNoAL', 'dav', '1';
update txx set c3 = '10', c1 = 'BxTXbyKRFBGbcPmPR' where c2 in ('dav', 'dav');
admin check table txx;
```

### 2. What did you expect to see? (Required)

success

### 3. What did you see instead (Required)

admin check fail

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Issue Number: close #23646 

Problem Summary:

### What is changed and how it works?

What's Changed, How it Works:

decode prefix column from value instead of use the value inside handle

### Related changes

- Need to cherry-pick to the release branch 5.0

### Check List 

Tests 

- Unit test

Side effects

- n/a

### Release note 

- No release note

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -3863,7 +3863,7 @@ func NewRowDecoder(ctx sessionctx.Context, schema *expression.Schema, tbl *model
 	if len(pkCols) == 0 {
 		pkCols = tables.TryGetCommonPkColumnIds(tbl)
 		if len(pkCols) == 0 {
-			pkCols = []int64{0}
+			pkCols = []int64{-1}
```
Comment (lysu) [inline executor/builder.go:3866]: -1 the value before clustered index for this code path...it must be a mistake to change this, so revert to -1

Comment (jackysp) [reply executor/builder.go:3866]: related to this bug fix?

Comment (lysu) [reply executor/builder.go:3866]: it seems cannot find a sql to reach here...but maybe we shouldn't change this..- -

File: executor/point_get.go [hunk 2]
```diff
@@ -544,6 +546,18 @@ func tryDecodeFromHandle(tblInfo *model.TableInfo, schemaColIdx int, col *expres
 	return false, nil
 }
 
+func notPKPrefixCol(colID int64, prefixColIDs []int64) bool {
+	if len(prefixColIDs) == 0 {
```
Comment (wjhuang2016) [inline executor/point_get.go:550]: These 3 lines are not needed.

**Pull request merged**